lib/extras/dec/jxl.cc

```diff
@@ -288,6 +288,30 @@ bool DecodeImageJXL(const uint8_t* bytes, size_t bytes_size,
         fprintf(stderr, "SelectFormat failed\n");
         return false;
       }
+      // Update selected format to forced bit depth.
+      // if we don't set the bit depth here, the JxlPixelFormat
+      // will match the tagged format, regardless of what bit
+      // depth the user requests, which will fail verification
+      switch (dparams.output_bitdepth.bits_per_sample) {
+        case 0:
+          // output_bitdepth == 0 means default
+          break;
+        case 8:
+          format.data_type = JXL_TYPE_UINT8;
+          break;
+        case 16:
+          format.data_type = JXL_TYPE_UINT16;
```

Comment:
Isn't float16 a possibility as well?

Reply:
The user doesn't have a way to request float16 on the command line, they only get a bits per sample value. None of the output formats support float16 that don't also support int16 so this is safer.

Reply:
> The user doesn't have a way to request float16 on the command line

Ah, therein lies the bug perhaps.



> None of the output formats support float16 that don't also support int16 so this is safer.

I beg to differ: [uint16 will clip, so it is less "safe"](https://github.com/libjxl/libjxl/blob/99b0721251b51028927c56a81657cf76de7ea320/lib/include/jxl/codestream_header.h#L167-L185); float16 is very popular/default format for EXRs that are used a lot in ACES workflows (movies, CGI, VFX...).

Reply:
The bug is that the option is not respected. Ideally the user should have an option to request the pixel format, but at the very least the existing option of bits-per-sample should actually work.